aptos-move/e2e-move-tests/src/tests/smart_data_structures.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, tests::common, MoveHarness};
+use aptos_types::account_address::AccountAddress;
+use aptos_package_builder::PackageBuilder;
+use crate::tests::gas::print_gas_cost;
+
+/// Run with `cargo test test_smart_data_structures_gas -- --nocapture` to see output.
+#[test]
+fn test_smart_data_structures_gas() {
+    let mut h = MoveHarness::new();
+    let fx_acc = h.aptos_framework_account();
+    let move_stdlib = common::framework_dir_path("move-stdlib");
+    let aptos_stdlib = common::framework_dir_path("aptos-stdlib");
+    assert_success!(h.publish_package(&fx_acc, &move_stdlib));
+    assert_success!(h.publish_package(&fx_acc, &aptos_stdlib));
+
+    let acc = h.new_account_at(AccountAddress::from_hex_literal("0xcafe").unwrap());
+    let mut builder = PackageBuilder::new("test");
+    builder.add_source(
+        "test",
+        "
+module 0xcafe::test {
+    use aptos_std::smart_vector::{Self, SmartVector};
+    use aptos_std::smart_table::{Self, SmartTable};
+    use aptos_std::table::{Self, Table};
+
+    struct SmartVectorStore has key {
+        v: SmartVector<u64>
+    }
+    struct VectorStore has key {
+        v: vector<u64>
+    }
+    struct SmartTableStore has key {
+        t: SmartTable<u64, u64>
+    }
+    struct TableStore has key {
+        t: Table<u64, u64>
+    }
+
+    public entry fun run_smart_vector(acct: &signer) {
+        let v = smart_vector::empty();
+        let i: u64 = 0;
+        while (i < 10000) {
+            smart_vector::push_back(&mut v, i);
+            i = i + 1;
+        };
+        move_to(acct, SmartVectorStore { v });
+    }
+
+    public entry fun run_vector(acct: &signer) {
+        let v = std::vector::empty();
+        let i: u64 = 0;
+        while (i < 10000) {
+            std::vector::push_back(&mut v, i);
+            i = i + 1;
+        };
+        move_to(acct, VectorStore { v });
+    }
+
+    public entry fun run_smart_table(acct: &signer) {
+        let t = smart_table::new();
+        let i: u64 = 0;
+        while (i < 1000) {
+            smart_table::add(&mut t, i, i);
```

Comment:
I think it might make more sense to break the cost for smart vector/table into 2 cases - when a new bucket is created and when an existing bucket is modified

Reply:
Is 1000 enough to lead to bucket split?

Reply:
each transaction will only have one write_set generated, so unless we can calculate total gas of 1000 txns, otherwise it make little difference.

Reply:
> Is 1000 enough to lead to bucket split?

it starts with 1 bucket and the first split happen at around the 80-th insertion.